### ISSUE TITLE
cant setup HA with external store (bootstrap data already found and encrypted with different token)

### ISSUE THREAD

**Issue by rowa78** [PyratLabs/ansible-role-k3s#137]
### Summary

I try to setup a cluster with HA and with an external datastore (mysql). The first control-pane node works fine. The second and third node says "starting kubernetes: preparing server: bootstrap data already found and encrypted with different token" in logfile.

### Issue Type

- Bug Report

### Controller Environment and Configuration

[pyratlabs-issue-dump.txt](https://github.com/PyratLabs/ansible-role-k3s/files/6871651/pyratlabs-issue-dump.txt)

```text
roles:
  # Install a role from Ansible Galaxy.
  - name: xanmanning.k3s
    version: v2.10.3
```

### Steps to Reproduce

hosts.yaml: 
```yaml
k3s_cluster:    
  hosts:
    pi-0:
      ansible_host: 192.168.0.40
      k3s_control_node: true
    pi-1:
      ansible_host: 192.168.0.41
      k3s_control_node: true
    pi-2:
      ansible_host: 192.168.0.42
      k3s_control_node: true    
    pi-3:
      ansible_host: 192.168.0.43
    vm-0:
      ansible_host: 192.168.0.39        
```

Playbook:
```yaml
---
- name: setup k3s-rancher
  hosts: "k3s_cluster"  
  vars:
    k3s_state: installed
    k3s_become_for_all: true
    k3s_server:
      datastore-endpoint: mysql://k3s:{{ datastore_password }}@tcp(192.168.0.3:3307)/k3s  
#    k3s_etcd_datastore: true
  roles:
    - xanmanning.k3s    
```

the host pi-0 works fine. In the logfile of pi-1 and pi-2 i get theese errors:

```
Jul 24 00:27:08 pi-2 k3s[72551]: time="2021-07-24T00:27:08.403901174Z" level=fatal msg="starting kubernetes: preparing server: bootstrap data already found and encrypted with different token"
Jul 24 00:27:08 pi-2 systemd[1]: k3s.service: Main process exited, code=exited, status=1/FAILURE
Jul 24 00:27:08 pi-2 systemd[1]: k3s.service: Failed with result 'exit-code'.
```

### Expected Result

```text
The cluster works with 3 control-panes and 2 workers.
```

### Actual Result

Errors in 2 control-panes
```text
Jul 24 00:26:59 pi-2 systemd-udevd[934]: Network interface NamePolicy= disabled on kernel command line, ignoring.
Jul 24 00:26:59 pi-2 k3s[72551]: time="2021-07-24T00:26:59Z" level=info msg="Acquiring lock file /var/lib/rancher/k3s/data/.lock"
Jul 24 00:26:59 pi-2 k3s[72551]: time="2021-07-24T00:26:59Z" level=info msg="Preparing data dir /var/lib/rancher/k3s/data/d57e75cb49c3cfd88307a8669e8adcf6b7740b66d6125a45c00aaa54301a5746"
Jul 24 00:27:07 pi-2 k3s[72551]: time="2021-07-24T00:27:07.309319662Z" level=info msg="Starting k3s v1.21.3+k3s1 (1d1f220f)"
Jul 24 00:27:07 pi-2 k3s[72551]: time="2021-07-24T00:27:07.327117363Z" level=info msg="Configuring mysql database connection pooling: maxIdleConns=2, maxOpenConns=0, connMaxLifetime=0s"
Jul 24 00:27:07 pi-2 k3s[72551]: time="2021-07-24T00:27:07.327324229Z" level=info msg="Configuring database table schema and indexes, this may take a moment..."
Jul 24 00:27:07 pi-2 k3s[72551]: time="2021-07-24T00:27:07.331178856Z" level=info msg="Database tables and indexes are up to date"
Jul 24 00:27:07 pi-2 k3s[72551]: time="2021-07-24T00:27:07.338109533Z" level=info msg="Kine listening on unix://kine.sock"
Jul 24 00:27:07 pi-2 k3s[72551]: time="2021-07-24T00:27:07.383731067Z" level=info msg="certificate CN=system:admin,O=system:masters signed by CN=k3s-client-ca@1627086427: notBefore=2021-07-24 00:27:07 +0000 UTC notAfter=2022-07-24 00:27:07 +0000 UTC"
Jul 24 00:27:07 pi-2 k3s[72551]: time="2021-07-24T00:27:07.386642602Z" level=info msg="certificate CN=system:kube-controller-manager signed by CN=k3s-client-ca@1627086427: notBefore=2021-07-24 00:27:07 +0000 UTC notAfter=2022-07-24 00:27:07 +0000 UTC"
Jul 24 00:27:07 pi-2 k3s[72551]: time="2021-07-24T00:27:07.389228717Z" level=info msg="certificate CN=system:kube-scheduler signed by CN=k3s-client-ca@1627086427: notBefore=2021-07-24 00:27:07 +0000 UTC notAfter=2022-07-24 00:27:07 +0000 UTC"
Jul 24 00:27:07 pi-2 k3s[72551]: time="2021-07-24T00:27:07.391797610Z" level=info msg="certificate CN=kube-apiserver signed by CN=k3s-client-ca@1627086427: notBefore=2021-07-24 00:27:07 +0000 UTC notAfter=2022-07-24 00:27:07 +0000 UTC"
Jul 24 00:27:07 pi-2 k3s[72551]: time="2021-07-24T00:27:07.394563000Z" level=info msg="certificate CN=system:kube-proxy signed by CN=k3s-client-ca@1627086427: notBefore=2021-07-24 00:27:07 +0000 UTC notAfter=2022-07-24 00:27:07 +0000 UTC"
Jul 24 00:27:07 pi-2 k3s[72551]: time="2021-07-24T00:27:07.396961248Z" level=info msg="certificate CN=system:k3s-controller signed by CN=k3s-client-ca@1627086427: notBefore=2021-07-24 00:27:07 +0000 UTC notAfter=2022-07-24 00:27:07 +0000 UTC"
Jul 24 00:27:07 pi-2 k3s[72551]: time="2021-07-24T00:27:07.399757266Z" level=info msg="certificate CN=k3s-cloud-controller-manager signed by CN=k3s-client-ca@1627086427: notBefore=2021-07-24 00:27:07 +0000 UTC notAfter=2022-07-24 00:27:07 +0000 UTC"
Jul 24 00:27:07 pi-2 k3s[72551]: time="2021-07-24T00:27:07.403929405Z" level=info msg="certificate CN=kube-apiserver signed by CN=k3s-server-ca@1627086427: notBefore=2021-07-24 00:27:07 +0000 UTC notAfter=2022-07-24 00:27:07 +0000 UTC"
Jul 24 00:27:07 pi-2 k3s[72551]: time="2021-07-24T00:27:07.408118859Z" level=info msg="certificate CN=system:auth-proxy signed by CN=k3s-request-header-ca@1627086427: notBefore=2021-07-24 00:27:07 +0000 UTC notAfter=2022-07-24 00:27:07 +0000 UTC"
Jul 24 00:27:07 pi-2 k3s[72551]: time="2021-07-24T00:27:07.412257943Z" level=info msg="certificate CN=etcd-server signed by CN=etcd-server-ca@1627086427: notBefore=2021-07-24 00:27:07 +0000 UTC notAfter=2022-07-24 00:27:07 +0000 UTC"
Jul 24 00:27:07 pi-2 k3s[72551]: time="2021-07-24T00:27:07.414697506Z" level=info msg="certificate CN=etcd-client signed by CN=etcd-server-ca@1627086427: notBefore=2021-07-24 00:27:07 +0000 UTC notAfter=2022-07-24 00:27:07 +0000 UTC"
Jul 24 00:27:07 pi-2 k3s[72551]: time="2021-07-24T00:27:07.418720036Z" level=info msg="certificate CN=etcd-peer signed by CN=etcd-peer-ca@1627086427: notBefore=2021-07-24 00:27:07 +0000 UTC notAfter=2022-07-24 00:27:07 +0000 UTC"
Jul 24 00:27:08 pi-2 k3s[72551]: time="2021-07-24T00:27:08.371950033Z" level=info msg="certificate CN=k3s,O=k3s signed by CN=k3s-server-ca@1627086427: notBefore=2021-07-24 00:27:07 +0000 UTC notAfter=2022-07-24 00:27:08 +0000 UTC"
Jul 24 00:27:08 pi-2 k3s[72551]: time="2021-07-24T00:27:08.373078029Z" level=info msg="Active TLS secret  (ver=) (count 8): map[listener.cattle.io/cn-10.43.0.1:10.43.0.1 listener.cattle.io/cn-127.0.0.1:127.0.0.1 listener.cattle.io/cn-192.168.0.42:192.168.0.42 listener.cattle.io/cn-kubernetes:kubernetes listener.cattle.io/cn-kubernetes.default:kubernetes.default listener.cattle.io/cn-kubernetes.default.svc:kubernetes.default.svc listener.cattle.io/cn-kubernetes.default.svc.cluster.local:kubernetes.default.svc.cluster.local listener.cattle.io/cn-localhost:localhost listener.cattle.io/fingerprint:SHA1=BE502296CDEC1053EA49A7C33D5FA3107BDEB703]"
Jul 24 00:27:08 pi-2 k3s[72551]: time="2021-07-24T00:27:08.403901174Z" level=fatal msg="starting kubernetes: preparing server: bootstrap data already found and encrypted with different token"
Jul 24 00:27:08 pi-2 systemd[1]: k3s.service: Main process exited, code=exited, status=1/FAILURE
Jul 24 00:27:08 pi-2 systemd[1]: k3s.service: Failed with result 'exit-code'.
Jul 24 00:27:08 pi-2 systemd[1]: Failed to start Lightweight Kubernetes
```

if i comment out the "datastore-endpoint" and use the "k3s_etcd_datastore" everything works fine?

Do you have an idea, what i am doing wrong?

**Comment by vrga** [comment]
from what i can tell its a change in behaviour with the newest k3s release https://github.com/k3s-io/k3s/releases/tag/v1.21.3%2Bk3s1

**Comment by xanmanning** [comment]
^^ Good shout. I hadn't seen this coming. Urgh.

We do [specify `--token-file`](https://github.com/PyratLabs/ansible-role-k3s/blob/main/templates/k3s.service.j2#L33), but it looks like they have made this explicitly needing `--token`?

Looking into it

**Comment by xanmanning** [comment]
Hopefully it will be fixed in v2.10.4 which will be available on Galaxy shortly.

**Comment by rowa78** [comment]
works perfectly now! Thank you:

k3s_cluster.yaml:
```
k3s_state: installed
k3s_release_version: v1.20.9+k3s1
k3s_become_for_all: yes
k3s_build_cluster: true

k3s_server:
  datastore-endpoint: mysql://k3s:{{ datastore_password }}@tcp(192.168.0.3:3307)/k3s  

k3s_server_manifests_urls:
  - url: https://kube-vip.io/manifests/rbac.yaml
    filename: kube-vip-rbac.yaml

k3s_server_manifests_templates:
  - "manifests/kube-vip/kube-vip-daemonset.yaml"

k3s_registration_address: 192.168.0.119

```


```
root@pi-0:/home/ubuntu# kubectl get nodes
NAME   STATUS   ROLES                  AGE    VERSION
pi-2   Ready    control-plane,master   51s    v1.20.9+k3s1
pi-0   Ready    control-plane,master   110s   v1.20.9+k3s1
pi-1   Ready    control-plane,master   53s    v1.20.9+k3s1
vm-0   Ready    <none>                 14s    v1.20.9+k3s1
pi-3   Ready    <none>                 10s    v1.20.9+k3s1
```
